lib/pos_blockchain.go

```diff
@@ -12,23 +12,67 @@ import (
 	"github.com/pkg/errors"
 )
 
+// processHeaderPoS validates and stores an incoming block header to build
```

Comment:
@lazynina, does this lightweight algorithm for building the PoS header chain seem reasonable? Once we download headers, we need to be able to build the header chain as a template for the block chain.

The node only syncs from outbound peers that it trusts, so it seems fine to optimistically store a block node for a header in the DB.